CRM/Activity/Import/Parser/Activity.php

```diff
@@ -264,7 +264,7 @@ public function import($onDuplicate, &$values) {
     $session = CRM_Core_Session::singleton();
     $dateType = $session->get('dateTypes');
     if (!isset($params['source_contact_id'])) {
-      $params['source_contact_id'] = $session->get('userID');
+      $params['source_contact_id'] = CRM_Core_Session::singleton()->getLoggedInContactID();
```

Comment:
IMHO, it's OK. We're already call `::singleton()` functions everywhere, and these tend to be pretty quick (just a `NULL`-check and `return`). I wouldn't do it inside some raster algorithm or low-level interrupt-handler... but for anything else, it's a drop in the bucket.